### PR TITLE
feat: Add "Privacy Policy" and "Terms of Service" link in footer

### LINKED ISSUE #1187: Add "Privacy Policy" and "Terms of Service" link in footer
## Is your enhancement request related to a problem? Please describe

Currently, only the signup and login pages have this link. There needs to be a global, easily accessible entrance.

## Describe the solution you'd like

Add links in the footer.

- Terms of Service: `/tos`
- Privacy: `/privacy`

image

### PR DESCRIPTION
**Pull request by Mindslayer001**
### Changes
- Added "Privacy Policy" (`/privacy`) and "Terms of Service" (`/tos`) links to the footer, accessible globally.
- Updated footer text to use i18n translation keys (`privacy`, `terms_of_service`, and `build_on`) for localization.
- Updated `i18n/en_US.yaml` to include the new keys.

### Testing
- Verified that the footer displays properly on all pages.
- Checked links to ensure they redirect to the appropriate pages.

#### Before:
Screenshot from 2024-12-30 13-48-50

#### After:
Screenshot from 2024-12-30 13-48-58

### Issue Reference
Closes #1187

### CODE REVIEW

**Comment by Mindslayer001** [comment]
Hi @sy-records and @shuashuai, I have made the changes based on your feedback. Could you please review the PR and let me know if there are any further improvements needed?

**Comment by sy-records** [comment]
Replace the locations of `/privacy` and `/tos` in this file to maintain the same ordering.

```js
          <NavLink to="/privacy" key="privacy" className="nav-link">
            {t('privacy')}
          </NavLink>
          <NavLink to="/tos" key="tos" className="nav-link">
            {t('tos')}
          </NavLink>
```

https://github.com/apache/incubator-answer/blob/3d2187d8491d582a41481e8a52f142e2750e500e/ui/src/pages/Legal/index.tsx#L34-L39

File: ui/src/components/Footer/index.tsx
Comment (sy-records) [inline ui/src/components/Footer/index.tsx:44]: I'm not sure we can reuse these translations?

```suggestion
          <Link to="/privacy" className="me-3">
            {t('label', { keyPrefix: 'admin.legal.privacy_policy' })}
          </Link>
          {/* Link to Terms of Service with right margin */}
          <Link to="/tos">
            {t('label', { keyPrefix: 'admin.legal.terms_of_service' })}
          </Link>
```

**Comment by Mindslayer001** [comment]
@sy-records  Thanks for feedback I will change according to the feedbacks and commit in few minutes.

**Comment by Mindslayer001** [comment]
Hi @sy-records , I have made the changes based on our previous discussions and feedback. Could you please review the PR and let me know if there are any further improvements needed?

File: ui/src/components/Footer/index.tsx
Comment (sy-records) [inline ui/src/components/Footer/index.tsx:40]: class should be `me-2`

File: ui/src/components/Footer/index.tsx
Comment (sy-records) [inline ui/src/components/Footer/index.tsx:51]:
```suggestion
            Powered by
            {/* eslint-disable-next-line react/jsx-no-target-blank */}
```

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx:51]: it will provide a space between "Power by" and "Apache Answer"

without {' '}:
Screenshot from 2024-12-31 10-32-24


 with {' '}:
Screenshot from 2024-12-31 10-32-12

Comment (sy-records) [reply ui/src/components/Footer/index.tsx:51]: But it fails to replace it when using the built file.
image

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx:51]: I didnt get it. are u saying that if I bilt it with "make ui" it is not working.

Comment (sy-records) [reply ui/src/components/Footer/index.tsx:51]: yep, 

```
make ui

go run cmd/answer/main.go run -C ./answer-data
```

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx:51]: I  got no file error "/home/mindslayer001/Project/incubator-answer
config file path:  answer-data/conf/config.yaml
Answer is starting..........................
panic: stat answer-data/conf/config.yaml: no such file or directory"

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx:51]: how about using old method of adding space 
`<a 
href="https://answer.apache.org" 
target="_blank" 
rel="noopener noreferrer">
&nbsp; Apache Answer
</a>`

Comment (sy-records) [reply ui/src/components/Footer/index.tsx:51]: Why are you adding spaces?
The original is the correct result

```
Powered by <a href="https://answer.apache.org" target="_blank"> Apache Answer </a>- the open-source software that powers Q&amp;A communities.<br>Made with love © 2024 Local Answer.
```

Comment (shuashuai) [reply ui/src/components/Footer/index.tsx:51]: > how about using old method of adding space `<a href="https://answer.apache.org" target="_blank" rel="noopener noreferrer"> &nbsp; Apache Answer </a>`

No need to adjust like this, just keep it the same as before.
image

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx:51]: It is because when I am running `npm run lint` the space between `powered by` and  `Apache` in removed.
So for Standardization purpose I am trying to add space.

Comment (sy-records) [reply ui/src/components/Footer/index.tsx:51]: okay, But you didn't solve the problem. https://github.com/apache/incubator-answer/pull/1217/files#r1899921473

Comment (Mindslayer001) [reply ui/src/components/Footer/index.tsx]: @sy-records @shuashuai I made the changes as per your feedback please check the pr

File: Makefile
Comment (sy-records) [inline Makefile:51]:
```suggestion
	@cd ui && pnpm pre-install && pnpm build && cd -
```

**Review by sy-records — CHANGES_REQUESTED** [review]

**Comment by Mindslayer001** [comment]
@sy-records I made the changes in footer/index.tsx to as requested and made the makefile to original

**Review by sy-records — APPROVED** [review]

**Comment by Mindslayer001** [comment]
Thanks guys for guiding me. I am beginner to opensource contributions.  @sy-records @shuashuai

**Comment by Mindslayer001** [comment]
Will my profile be visible on repo contributors list.
image

**Comment by shuashuai** [comment]
> Will my profile be visible on repo contributors list. image

Wait until your code is merged into the main branch, specifically until version 1.4.5 is released

**Pull request merged**
